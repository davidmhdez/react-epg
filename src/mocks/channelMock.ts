import Channel, {ChannelEvent} from "../domain/entity/Channel.ts";
import ProgramGuide from "../domain/entity/ProgramGuide.ts";

export const channelEventMock: ChannelEvent = {
    id: "11",
    name: "Test event",
    duration: "01:30:00",
    dateBegin: new Date(2024, 6, 25, 14, 0, 0),
    description: "Test description",
    durationInMinutes: 90,
    dateEnd: new Date(2024, 6, 25, 15, 30, 0),
}

export const channelMock: Channel = {
    id: "1",
    name: "Test channel",
    number: "3",
    image: "https://www.testimage.com/assets/image.png",
    events: [
        channelEventMock
    ]
}

export const programGuideMock: ProgramGuide = {
    from: new Date(2024, 6, 25, 14, 0, 0),
    end: new Date(2024, 6, 25, 23, 59, 59),
    channels: [
        channelMock
    ]
}