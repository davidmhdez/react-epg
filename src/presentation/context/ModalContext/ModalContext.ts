import {createContext} from "react";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type ModalContextValues = {
    isOpen: boolean
    openModal: () => void
    closeModal: () => void
    selectedChannelEvent: ChannelEvent | null
    selectChannelEvent: (channelEvent: ChannelEvent) => void
}

export const initialModalContextValues = {
    isOpen: false,
    selectedChannelEvent: null,
    closeModal: () => {},
    openModal: () => {},
    selectChannelEvent: () => {},
}

const ModalContext = createContext<ModalContextValues>(initialModalContextValues)

export default ModalContext