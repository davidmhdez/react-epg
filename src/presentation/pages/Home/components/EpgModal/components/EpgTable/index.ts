export {default} from "./EpgTable.tsx"