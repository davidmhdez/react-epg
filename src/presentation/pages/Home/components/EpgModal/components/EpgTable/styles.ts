import styled from "styled-components";

export const EpgTableWrapper = styled.div`
    width: 100%;
    position: relative;
`

export const EpgTableBody = styled.div`
    position: relative;
    display: flex;
    flex-direction: column;
    gap: 10px;
    
`