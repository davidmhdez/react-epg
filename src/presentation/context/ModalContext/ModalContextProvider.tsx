import React, {ReactNode, useState} from "react";
import ModalContext from "./ModalContext.ts";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type Props = {
    children: ReactNode
}

const ModalContextProvider: React.FC<Props> = ({children}) => {
    const [isOpen, setIsOpen] = useState(false);
    const [selectedChannelEvent, setSelectedChannelEvent] = useState<ChannelEvent | null>(null)
    const openModal = () => {
        setIsOpen(true);
        setSelectedChannelEvent(null);
    }
    const closeModal = () => {
        setIsOpen(false);
        setSelectedChannelEvent(null);
    }

    const selectChannelEvent = (channelEvent: ChannelEvent) => setSelectedChannelEvent(channelEvent);

    return (
        <ModalContext.Provider value={{
            isOpen,
            selectedChannelEvent,
            selectChannelEvent,
            openModal,
            closeModal
        }}>
            {children}
        </ModalContext.Provider>
    );
};

export default ModalContextProvider;
