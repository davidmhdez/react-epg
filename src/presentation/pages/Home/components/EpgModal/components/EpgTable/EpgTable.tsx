import React from "react";
import {EpgTableBody, EpgTableWrapper} from "./styles.ts";
import ProgramGuide from "../../../../../../../domain/entity/ProgramGuide.ts";
import EpgChannelEventRow from "./components/EpgChannelEventRow/EpgChannelEventRow.tsx";
import EpgHoursHeader from "./components/EpgHoursHeader";
import {getWidthFromDates} from "../../helpers.ts";

type Props = {
    programGuide: ProgramGuide
}

const EpgTable: React.FC<Props> = React.memo(({programGuide}) => {
    const maxTableWidth = getWidthFromDates(programGuide.from, programGuide.end);

    return (
        <EpgTableWrapper>
            <EpgHoursHeader startDate={programGuide.from} endDate={programGuide.end}/>
            <EpgTableBody>
                {programGuide.channels.map(channel =>(
                    <EpgChannelEventRow
                        channel={channel}
                        key={channel.id}
                        programGuideBegin={programGuide.from}
                        maxWidth={maxTableWidth}
                    />
                ))}
            </EpgTableBody>
        </EpgTableWrapper>
    );
});

export default EpgTable;
