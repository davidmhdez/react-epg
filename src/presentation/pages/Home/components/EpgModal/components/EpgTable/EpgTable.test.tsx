import {render} from "@testing-library/react";
import ModalContext, {initialModalContextValues} from "../../../../../../context/ModalContext/ModalContext.ts";
import EpgTable from "./EpgTable.tsx";
import {programGuideMock} from "../../../../../../../mocks/channelMock.ts";

describe('EpgTable test', () => {
    test('should match with snapshot', () => {
        const { container } = render(
            <ModalContext.Provider value={initialModalContextValues}>
                <EpgTable programGuide={programGuideMock}/>
            </ModalContext.Provider>
        )
        expect(container).toMatchSnapshot()
    })
})