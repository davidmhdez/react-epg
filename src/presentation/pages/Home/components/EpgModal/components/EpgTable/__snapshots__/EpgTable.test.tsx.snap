// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpgTable test should match with snapshot 1`] = `
<div>
  <div
    class="sc-aYaIB hgNKVS"
    data-testid="epgTable"
  >
    <div
      class="sc-kpDprT bEUryy"
    >
      <p
        class="sc-dAlxHm kVScoi"
      >
        HOY
      </p>
      <div
        class="sc-jlZhRR jGcWDG"
      >
        <div
          class="sc-cwHqhk fsfldu"
        >
          14:00
        </div>
        <div
          class="sc-cwHqhk curgWK"
        >
          14:30
        </div>
        <div
          class="sc-cwHqhk jMXTpN"
        >
          15:00
        </div>
        <div
          class="sc-cwHqhk dxSojT"
        >
          15:30
        </div>
        <div
          class="sc-cwHqhk bibaaM"
        >
          16:00
        </div>
        <div
          class="sc-cwHqhk fiYASY"
        >
          16:30
        </div>
        <div
          class="sc-cwHqhk jMKLxn"
        >
          17:00
        </div>
        <div
          class="sc-cwHqhk eKwFUm"
        >
          17:30
        </div>
        <div
          class="sc-cwHqhk gHukmJ"
        >
          18:00
        </div>
        <div
          class="sc-cwHqhk eeySzd"
        >
          18:30
        </div>
        <div
          class="sc-cwHqhk hHkltO"
        >
          19:00
        </div>
        <div
          class="sc-cwHqhk loZGas"
        >
          19:30
        </div>
        <div
          class="sc-cwHqhk jaflJT"
        >
          20:00
        </div>
        <div
          class="sc-cwHqhk eHZgzP"
        >
          20:30
        </div>
        <div
          class="sc-cwHqhk kWRoYl"
        >
          21:00
        </div>
        <div
          class="sc-cwHqhk idBUT"
        >
          21:30
        </div>
        <div
          class="sc-cwHqhk hIZWAo"
        >
          22:00
        </div>
        <div
          class="sc-cwHqhk lhkIIQ"
        >
          22:30
        </div>
        <div
          class="sc-cwHqhk jOhkAT"
        >
          23:00
        </div>
        <div
          class="sc-cwHqhk ghbuyF"
        >
          23:30
        </div>
      </div>
    </div>
    <div
      class="sc-gEvDqW hipaoh"
    >
      <div
        class="sc-eqUzNf doxFpK"
      >
        <div
          class="sc-kAycRU innHW"
        >
          <div
            class="sc-imWZod gZXqPU"
          >
            <p
              class="sc-jXbVAB eRvmwg"
            >
              3
            </p>
            <img
              alt="Test channel logo"
              class="sc-dhKdPU egNOtl"
              src="https://www.testimage.com/assets/image.png"
            />
          </div>
        </div>
        <div
          class="sc-fqkwJk ctBQky"
        >
          <div
            class="sc-dcJtft kOXGtT"
            data-testid="11"
          >
            <h6
              class="sc-iGgVNO cFzOhe"
              title="Test event"
            >
              Test event
            </h6>
            <p
              class="sc-gsFSjX hziKuj"
              title="14:00 - 15:30"
            >
              14:00 - 15:30
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
